pkg/controller/releasemanager/plan/syncServiceLevelAlerts.go

```diff
@@ -0,0 +1,133 @@
+package plan
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/go-logr/logr"
+	picchuv1alpha1 "go.medium.engineering/picchu/pkg/apis/picchu/v1alpha1"
+	"go.medium.engineering/picchu/pkg/plan"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SyncServiceLevelAlerts struct {
+	App                         string
+	Namespace                   string
+	Labels                      map[string]string
+	ServiceLevelObjectiveLabels picchuv1alpha1.ServiceLevelObjectiveLabels
+	ServiceLevelObjectives      []*picchuv1alpha1.ServiceLevelObjective
+}
+
+func (p *SyncServiceLevelAlerts) Apply(ctx context.Context, cli client.Client, scalingFactor float64, log logr.Logger) error {
+	serviceLevelAlerts, err := p.ServiceLevelAlerts()
+	if err != nil {
+		return err
+	}
+
+	if len(serviceLevelAlerts) > 0 {
+		for _, pr := range serviceLevelAlerts {
```

Comment:
hi my name is Go

I hope you like copy–pasting

because it's all I brought to the party

Reply:
:lolsob: